### ISSUE TITLE
Invalid named parameter "maximum_depth" for function read_json

### ISSUE THREAD

**Issue by polyrand** [duckdb/duckdb#6929]
### What happens?

The JSON reading functions don't recognize the `maximum_depth` parameter.

### To Reproduce

Generate data:

```python
import json

data = [{"foo": 1, "bar": 2}, {"foo": 3, "bar": 4}]

with open("data.json", "w") as f:
    for record in data:
        f.write(json.dumps(record))
        f.write("\n")
```

Query:

```sh
duckdb -s "select * from read_json('data.json', maximum_depth=5)"
```

Error:

```
Error: Binder Error: Invalid named parameter "maximum_depth" for function read_json
```

### OS:

Tested on macOS and Ubuntu

### DuckDB Version:

v0.7.1 b00b93f0b1 or v0.7.2-dev1256 cef3bdcb8c

### DuckDB Client:

CLI / Python

### Full Name:

Ricardo Ander-Egg

### Affiliation:

Based on experiments run for personal usage

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Tishj** [comment]
To add to that ^
You said it didn't work for `read_json_auto` either, can you verify this?
Because for me it does work, for example using the python client:
```py
duckdb.sql("select * from read_json_auto('data.json', maximum_depth=5)")
```

**Comment by polyrand** [comment]
Yes, it works with `read_json_auto`. I was 100% sure I had tried it, but I was wrong! I guess I maybe just tried `read_json('data.json', autodetect=True, maximum_depth=5)` and forgot to also try replacing `autodetect=True` -> `read_json_auto`. Perhaps I incorrectly assumed that `read_json_auto()` was the same as `read_json(autodetect=True)`. In any case, this is working:

```sh
duckdb -c "select * from read_json_auto('data.json', maximum_depth=5)"

duckdb -c "select * from read_ndjson_auto('data.json', maximum_depth=5)"
```

I'll close the issue since it's been resolved. Thank you for the clarification!
